packages/builder/src/builderStore/store/index.js

```diff
@@ -572,10 +576,58 @@ const getPathToComponent = store => component => {
   return path
 }
 
+const generateNewIdsForComponent = component =>
+  walkProps(component, p => {
+    p._id = uuid()
+  })
+
+const storeComponentForCopy = store => (component, cut = false) => {
+  store.update(s => {
+    const copiedComponent = cloneDeep(component)
+    s.componentToPaste = copiedComponent
+    if (cut) {
+      const parent = getParent(s.currentPreviewItem.props, component._id)
+      parent._children = parent._children.filter(c => c._id !== component._id)
+      _selectComponent(s, parent)
+    }
+
+    return s
+  })
+}
+
+const pasteComponent = store => (targetComponent, mode) => {
```

Comment:
Having these functions exist at the very top level clutters this already huge file.

It would make more sense to co-locate these functions at the same level at the component tree itself, since that's the only place in the app they are used